Distraction Shield/modules/statistics/Tracker.js

```diff
@@ -74,10 +75,12 @@ export default class Tracker {
             if(tab) {
                 if (this.compareDomain(tab.url, constants.zeeguuExTracker)) {
                     this.incTimeExercises(timeSpent);
+                    this.incTotalTimeExercises(timeSpent);
```

Comment:
there is a small problem with this approach @ancasuliman. 

i've installed the new version of the extension and now
my total time is out of sync with the time spent... 

<img width="822" alt="screen shot 2019-02-09 at 11 50 01 am" src="https://user-images.githubusercontent.com/464519/52519294-c3cdb900-2c61-11e9-811e-788a3435935b.png">

i think the safer solution would be to compute the invested 
time and the wasted time at the point where you display them
since they are clearly *derived* quantities. 

this has the added benefit of, if later for example, you want to
compute invested time vs. wasted time per a specific interval
(e.g. this week, or this month) it would be straightforward. 

what do you think?

Reply:
@mircealungu the time values are recorded in milliseconds and the little differences occur at the moment of conversion (with approximation) from milliseconds to hh:mm:ss, conversion that is done at display. We could record values directly in the hh:mm:ss format. What do you think?

Reply:
In addition, since you mentioned that statistics could be organized based on week or month, we could add a dropdown menu with these values, so that the user could choose on their own which data to be displayed within the table. In this way, we will keep the statistics page simple and easy to read. What do you think about this?

Reply:
@ancasuliman - i wonder if there's a mis-understanding between us :) 
if you look at time wasted, and you sum up all those columns, it's not
a few milliseconds - it's a huge difference. this is because, the total time 
wasted started to be computed only since i deployed the new version,
however, the individual segments were stored since before... thus the 
current approach does not work when upgrading. do you see what i 
mean? or did you see it all along and i didn't understand what you said?

if you sum up milliseconds and then convert, or you convert and then 
sum up seconds, it's the same for me. whatever is the simplest.

Reply:
instead of dropdown we can simply add this info to the top box. 
if you look at it, it already provides info about 
- last 24h
- last week
- last month
- since installing the extension

<img width="836" alt="screen shot 2019-02-09 at 7 47 39 pm" src="https://user-images.githubusercontent.com/464519/52525010-94d83700-2ca3-11e9-84a9-8bf73caec732.png">

Reply:
@mircealungu I didn t look attentively on the picture you posted, I'm sorry. I had the same problem as the one you mentioned, I deleted the extension and installed it again and it worked fine, the old data was deleted and statistics started being recorded from the beginning. The fact is I don t quite understand why the intercept data is still available even if the extension is removed and loaded again (for a single time though, as I mentioned above), since the storage of the Chrome API is initialized with empty lists when init.js is run.

Reply:
Regarding the top box, if we add the total time statistics there, I would suggest changing it into a table which will display information about both exercise and intercept time during last week, last month etc. What do you think?

Reply:
no worries. 

but are we on the same page with the idea of 
computing the total time by summing up the individual 
quantities as opposed to tracking two more quantities? 

then for the formatting, a table would definitely be great!

Reply:
@mircealungu the first approach I wanted to apply was to sum up all the time values recorded for interceptions, respectively practicing, as you mentioned, but I found it more difficult to manipulate the data this way. This data is recorded in the storage at keys tds_blacklist and tds_exerciseTime, in milliseconds. In order to avoid obtaining that small delay, we would have to sum the values after they have already been converted to hh:mm:ss format. Also, the feature of identifying intercept times is already implemented (using a separate storage key), so, in order to implement it for language practice as well, I was thinking to add a different storage key for it.

Reply:
@ancasuliman - oh but do you really think that milliseconds rounding error will be a problem? from my POV you can round to the next integer minute and it would still be ok. at the end of a month time wasted and time invested should be in hours... thus some milliseconds and even seconds that are different won't be a big problem i would say. what do you think?

regarding this, I'm not sure i understand what do you plan to implement for language practice?

> Also, the feature of identifying intercept times is already implemented (using a separate storage key), so, in order to implement it for language practice as well, I was thinking to add a different storage key for it.